sed/calibrator/hextof.py

```diff
@@ -0,0 +1,154 @@
+        if config is None:
+            raise ValueError("Either sector_delays or config must be given.")
+        sector_delays = config["dataframe"]["sector_delays"]
+    if sector_id_column is None:
+        if config is None:
+            raise ValueError("Either sector_id_column or config must be given.")
+        sector_id_column = config["dataframe"]["sector_id_column"]
+    if tof_column is None:
+        if config is None:
+            raise ValueError("Either tof_column or config must be given.")
+        tof_column = config["dataframe"]["tof_column"]
+    # align the 8s sectors
+    sector_delays_arr = dask.array.from_array(sector_delays)
+
+    def align_sector(x):
+        val = x[tof_column] - sector_delays_arr[x[sector_id_column].values.astype(int)]
+        return val.astype(np.float32)
+    df[tof_column] = df.map_partitions(
+        align_sector, meta=(tof_column, np.float32)
+    )
+    metadata: Dict[str,Any] = {
+        "applied": True,
+        "sector_delays": sector_delays,
+    }
+    return df, metadata
+
+
+def dld_time_to_ns(
+        df: Union[pd.DataFrame, dask.dataframe.DataFrame],
+        tof_ns_column: str = None,
+        tof_binwidth: float = None,
+        tof_column: str = None,
+        tof_binning: int = None,
+        config: dict = None,
+) -> Tuple[Union[pd.DataFrame, dask.dataframe.DataFrame], dict]:
+    """Converts the 8s time in steps to time in ns.
+
+    Args:
+        tof_binwidth (float, optional): Time step size in nanoseconds.
+            Defaults to config["dataframe"]["tof_binwidth"].
+        tof_column (str, optional): Name of the column containing the
+            time-of-flight steps. Defaults to config["dataframe"]["tof_column"].
+        tof_column (str, optional): Name of the column containing the
+            time-of-flight. Defaults to config["dataframe"]["tof_column"].
+        tof_binning (int, optional): Binning of the time-of-flight steps.
+    
+    Returns:
+        dask.dataframe.DataFrame: Dataframe with the new columns.
+        dict: Metadata dictionary.
+    """
+    if tof_binwidth is None:
+        if config is None:
+            raise ValueError("Either tof_binwidth or config must be given.")
+        tof_binwidth = config["dataframe"]["tof_binwidth"]
+    if tof_column is None:
+        if config is None:
+            raise ValueError("Either tof_column or config must be given.")
+        tof_column = config["dataframe"]["tof_column"]
+    if tof_binning is None:
+        if config is None:
+            raise ValueError("Either tof_binning or config must be given.")
+        tof_binning = config["dataframe"]["tof_binning"]
+    if tof_ns_column is None:
+        if config is None:
+            raise ValueError("Either tof_ns_column or config must be given.")
+        tof_ns_column = config["dataframe"]["tof_ns_column"]
+
+    def convert_to_ns(x):
+        val = x[tof_column] * tof_binwidth * 2**tof_binning
+        return val.astype(np.float32)
```

Comment:
That is dangerous, as here you will actually lose accuracy!

Reply:
We explicitly pass tof as float64 here to net lose precision, I tested this: https://github.com/OpenCOMPES/sed/blob/ec5bccda4f917f960226c1b0d9bb90d45c422a58/sed/calibrator/energy.py#L859

Reply:
if you have tested this and found a case in which you loose precision, I trust you.  So far I never noticed a loss in precision with 32 bit but I might not have checked that thorouhgly.